File: code-of-might-and-magic-go/code-of-might-and-magic/print.go
package main

import (
	"fmt"
	"os"
)

func debug(message string, a ...interface{}) {
	fmt.Fprintln(os.Stderr, fmt.Sprintf(message, a...))
}

func pick(index int) {
	fmt.Println(fmt.Sprintf("PICK %d", index))
}

func summon(card creature) {
	fmt.Print(fmt.Sprintf("SUMMON %d;", card.instanceID))
}

func attack(card creature, target int) {
	fmt.Print(fmt.Sprintf("ATTACK %d %d;", card.instanceID, target))
}

File: code-of-might-and-magic-kotlin/src/Print.kt
fun debug(message: String) {
	System.err.println(message)
}

fun pick(index: Int) {
	println("PICK $index")
}

fun attack(card: Card, target: Int): String {
	return "ATTACK ${card.instanceID} $target"
}

fun summon(card: Card): String {
	return "SUMMON ${card.instanceID}"
}

fun use(card: Card, target: Int): String {
	return "USE ${card.instanceID} $target"
}